### PR TITLE
Added a PolicyBot for python

### PR DESCRIPTION
**Pull request by aliyakamercan**
- Added a bot that takes a policy. 
- added `open_spiel::MakePolicyBot` to pyspiel, had to change a few policy `unique_ptr`s to `shared_ptr` for this.

### CODE REVIEW

**Comment by inejc** [comment]
Hi @aliyakamercan, I believe your changes also address what is described in https://github.com/deepmind/open_spiel/pull/240. If your PR is merged before https://github.com/deepmind/open_spiel/pull/240, we could add the missing python bindings for MCCFR solvers.

**Comment by inejc** [comment]
Note that there are python-exposed raw pointers to `Policy` in function arguments, e.g. [here](https://github.com/deepmind/open_spiel/blob/d2a8054a75e8ab577244b28fca65eae3d5c1ec85/open_spiel/python/pybind11/policy.cc#L49). I'm not completely sure whether the issue described in https://pybind11.readthedocs.io/en/stable/advanced/smart_ptrs.html#std-shared-ptr applies but we should probably check this.

**Comment by lanctot** [comment]
Hi guys, going to be merging #240 in the next hour. Does it affect this PR / overlap?

**Comment by lanctot** [comment]
Hmm, I'm surprised changing this many pointers is necessary for this. @elkhrt , any opinion on this?

**Comment by inejc** [comment]
> Does it affect this PR / overlap?

The reason some constructors were not exposed in https://github.com/deepmind/open_spiel/pull/240 is that they expect `Policy` via a shared ptr and this PR changes the holder type of `Policy` from unique to a shared ptr so the two PRs overlap to some extent.

**Comment by aliyakamercan** [comment]
Hi @inejc, @lanctot 

> Hi @aliyakamercan, I believe your changes also address what is described in #240. If your PR is merged before #240, we could add the missing python bindings for MCCFR solvers. cc @lanctot

I think it addresses some but not all of them. It should be easy to change the rest.

> Hmm, I'm surprised changing this many pointers is necessary for this. @elkhrt , any opinion on this?

https://pybind11.readthedocs.io/en/stable/advanced/smart_ptrs.html#std-unique-ptr

It's not possible to use unique pointers as function args.  Changing the signature for `MakePolicyBot` required all the other changes.

File: open_spiel/python/algorithms/evaluate_bots_test.py [hunk 1]
```diff
@@ -18,25 +18,39 @@
 from __future__ import division
 from __future__ import print_function
 
-from absl.testing import absltest
+from absl.testing import absltest, parameterized
```
Comment (finbarrtimbers) [inline open_spiel/python/algorithms/evaluate_bots_test.py:21]: Please put this on a separate line.

**Comment by finbarrtimbers** [comment]
Hey, this looks good to me, but I think julia_test is failing:

```
ERROR: LoadError: InitError: Unregistered method with key (:function, :new_initial_state, :OpenSpiel, 0x42c9f5e7c174ee9e) requested, maybe you need to precompi
le the Julia module?                                                                                                                                           
Stacktrace:                                                                                                                                                    
 [1] error(::String) at ./error.jl:33
 [2] _get_function_pointer(::Tuple{Symbol,Symbol,Symbol,UInt64}) at /usr/local/google/home/finbarrtimbers/.julia/packages/CxxWrap/94t40/src/CxxWrap.jl:376
 [3] initialize_julia_module(::Module) at /usr/local/google/home/finbarrtimbers/.julia/packages/CxxWrap/94t40/src/CxxWrap.jl:414
 [4] __init__() at /usr/local/google/home/finbarrtimbers/tmp/open_spiel/open_spiel/julia/src/OpenSpiel.jl:24
 [5] _include_from_serialized(::String, ::Array{Any,1}) at ./loading.jl:692
 [6] _require_search_from_serialized(::Base.PkgId, ::String) at ./loading.jl:776
 [7] _require(::Base.PkgId) at ./loading.jl:1001
 [8] require(::Base.PkgId) at ./loading.jl:922
 [9] require(::Module, ::Symbol) at ./loading.jl:917
 [10] include at ./boot.jl:328 [inlined]
 [11] include_relative(::Module, ::String) at ./loading.jl:1105
 [12] include(::Module, ::String) at ./Base.jl:31
 [13] include(::String) at ./client.jl:424
 [14] top-level scope at none:6
during initialization of module OpenSpiel
in expression starting at /usr/local/google/home/finbarrtimbers/tmp/open_spiel/open_spiel/julia/test/runtests.jl:1
ERROR: Package OpenSpiel errored during testing
Stacktrace:
 [1] pkgerror(::String, ::Vararg{String,N} where N) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/Pkg/src/Types.jl:113
 [2] #test#131(::Bool, ::Nothing, ::Cmd, ::Cmd, ::typeof(Pkg.Operations.test), ::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /buildworker/worker/pa
ckage_linux64/build/usr/share/julia/stdlib/v1.3/Pkg/src/Operations.jl:1372
 [3] #test at ./none:0 [inlined]
 [4] #test#62(::Bool, ::Nothing, ::Cmd, ::Cmd, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(Pkg.API.test), ::Pkg.Types.Cont
ext, ::Array{Pkg.Types.PackageSpec,1}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/Pkg/src/API.jl:253
 [5] test(::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/Pkg/src/API.jl:239
 [6] #test#58 at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/Pkg/src/API.jl:233 [inlined]
 [7] test() at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/Pkg/src/API.jl:230
 [8] top-level scope at none:1
```

**Comment by lanctot** [comment]
Hi @finbarrtimbers I think this is unrelated to this PR. I ran into it as well and was able to fix it. I sent you a few messages internally, let me know if it helps.

**Comment by finbarrtimbers** [comment]
You were right- it's due to a separate issue. Sorry for the confusion!

**Comment by finbarrtimbers** [comment]
The code looks good- I'm merging this, and you'll get a notification in the next few days when it's in. This will most likely be on Monday.

**Pull request merged**